fbw-a32nx/src/systems/instruments/src/MCDU/legacy_pages/A320_Neo_CDU_IRSStatus.ts

```diff
@@ -1,87 +1,268 @@
-// @ts-strict-ignore
-import { CDUIRSMonitor } from './A320_Neo_CDU_IRSMonitor';
-import { CDUIRSStatusFrozen } from './A320_Neo_CDU_IRSStatusFrozen';
+// Copyright (c) 2021-2023, 2025 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+import { Arinc429Register, coordinateToString, RegisteredSimVar } from '@flybywiresim/fbw-sdk';
 import { LegacyFmsPageInterface } from '../legacy/LegacyFmsPageInterface';
+import { SimVarValueType } from '@microsoft/msfs-sdk';
+import { Coordinates } from '@fmgc/flightplanning/data/geo';
+import { FmsFormatters } from '../legacy/FmsFormatters';
 
 export class CDUIRSStatus {
-  static ShowPage(mcdu: LegacyFmsPageInterface, index, prev_wind_dir?) {
-    let currPos = new LatLong(
-      SimVar.GetSimVarValue('GPS POSITION LAT', 'degree latitude'),
-      SimVar.GetSimVarValue('GPS POSITION LON', 'degree longitude'),
-    ).toShortDegreeString();
-    let currPosSplit: string[];
-    let sep: string;
-    if (currPos.includes('N')) {
-      currPosSplit = currPos.split('N');
-      sep = 'N/';
-    } else {
-      currPosSplit = currPos.split('S');
-      sep = 'S/';
-    }
-    const latStr = currPosSplit[0];
-    const lonStr = currPosSplit[1];
-    currPos = latStr + sep + lonStr;
-    const GROUNDSPEED = SimVar.GetSimVarValue('GPS GROUND SPEED', 'Knots') || '0';
-    const THDG = SimVar.GetSimVarValue('GPS GROUND TRUE HEADING', 'radians') || '000';
-    const TTRK = SimVar.GetSimVarValue('GPS GROUND MAGNETIC TRACK', 'radians') || '000';
-    const MHDG = SimVar.GetSimVarValue('GPS GROUND TRUE TRACK', 'radians') || '000';
-    const WIND_VELOCITY = SimVar.GetSimVarValue('AMBIENT WIND VELOCITY', 'Knots') || '00';
-    // wind direction smoothing like A32NX_NDInfo.js:setWind()
-    let wind_dir = Math.round(Simplane.getWindDirection());
-    if (typeof prev_wind_dir == 'undefined') {
-      prev_wind_dir = wind_dir;
-    }
-    let startAngle = prev_wind_dir;
-    let endAngle = wind_dir;
-    const delta = endAngle - startAngle;
-    if (delta > 180) {
-      startAngle += 360;
-    } else if (delta < -180) {
-      endAngle += 360;
-    }
-    // FIXME filtering inside the page doesn't make a lot of sense!
-    const smoothedAngle = Utils.SmoothSin(startAngle, endAngle, 0.25, mcdu._deltaTime / 1000);
-    wind_dir = smoothedAngle % 360;
+  private static frozen = false;
+
+  private static readonly ir1Latitudevar = RegisteredSimVar.create<number>(
```

Comment:
Use camelCase for all variables here

```suggestion
  private static readonly ir1LatitudeVar = RegisteredSimVar.create<number>(
```